src/creator.rs

```diff
@@ -95,60 +98,241 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    fn check_ready(&self, round: Round) -> bool {
-        if round == 0 {
-            return true;
-        }
-        if round > self.max_round {
-            warn!(target: "AlephBFT-creator", "{:?} Maximum round reached. Not creating another unit.", self.node_ix);
-            return false;
+    async fn ready_for_create(&mut self, round: Round) {
+
+        let prev_round_index = match round.checked_sub(1) {
+            Some(prev_round) => prev_round as usize,
+            None => return,
+        };
+
+        let delay = time::sleep((self.create_lag)(round.into()));
+        tokio::pin!(delay);
+        loop {
+            if round < self.current_round {
+                return ();
+            }
+
+            tokio::select! {
+                unit = self.parents_rx.next() => {
+                    if let Some(u) = unit {
+                        self.add_unit(u.round(), u.creator(), u.hash());
+                    }
+                    continue;
+                }
+                _ = &mut delay => {
+                    break;
+                }
+            }
         }
+
         // To create a new unit, we need to have at least >floor(2*N/3) parents available in previous round.
         // Additionally, our unit from previous round must be available.
-        let prev_round_index = (round - 1) as usize;
         let threshold = (self.n_members * 2) / 3;
```

Comment:
throughout the repository, `threshold` is defined as `floor(2*N/3) + 1` that is the required number of parents. Can we change it for consistency?